src/Microsoft.DotNet.Build.Tasks/GetTargetArchitecture.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GetTargetArchitecture : Task
```

Comment:
Can we combine this and the GetTargetOS tasks? I don't think there is a much value in splitting them. We can call it GetTargetMachineInfo and have a TargetArch and TargetOS output property.